tests/lints/check-public-headers.py

```diff
@@ -29,6 +29,7 @@
     "chemfiles/config.h",
     "chemfiles/sorted_set.hpp",
     "chemfiles/unreachable.hpp",
+    "chemfiles/files/MemoryBuffer.hpp",
```

Comment:
I think we should be able to use a trick similar to the one used for Trajectory (`destructor = default` in the .cpp file) to fix the issue with unique_ptr destructor needing the full definition of MemoryBuffer. I'll check this locally.

Reply:
I don't think this will work. The 'Topology`, `UnitCell`, and `Format` classes are eventually defined in other public headers. The issue is that `unique_ptr` has no way of knowing the size of the `MemoryBuffer` object without the full class definition. Maybe there is a way around this, however.